### PR TITLE
SSE: Fix NoData when some series were no data but others not

### PR DESCRIPTION
**Pull request by kylebrandt**
**What this PR does / why we need it**:
Seems the logic was such that it considered the whole condition NoData if any of the series were Empty, but I think it should be only NoData if all the series in the condition are empty.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Over tired, could maybe use some more tests are thinking through the logic to make sure I have this correct....

### CODE REVIEW

**Comment by Dieterbe** [comment]
> Seems the logic was such that it considered the whole condition NoData if any of the series were Empty, but I think it should be only NoData if all the series in the condition are empty.

I think that is grafana alerts used to work.
With Graphite it's not uncommon to have some series with only null points, and other series that do have data.

**Comment by santihernandezc** [comment]
@Dieterbe so you're saying we should make this a Graphite only thing? Are there any other data sources that behave similarly?

**Comment by kylebrandt** [comment]
> > Seems the logic was such that it considered the whole condition NoData if any of the series were Empty, but I think it should be only NoData if all the series in the condition are empty.
> 
> I think that is grafana alerts used to work (?) With Graphite it's not uncommon to have some series with only null points, and other series that do have data.

I think probably so, it was an error on my part in translating the old alert behavior to into SSE.

Previous code that this was meant to match:

https://github.com/grafana/grafana/blob/33b6f996e1a24e22a3f54f16a65a14635e407fb6/pkg/services/alerting/eval_handler.go#L30

and 

https://github.com/grafana/grafana/blob/33b6f996e1a24e22a3f54f16a65a14635e407fb6/pkg/services/alerting/conditions/query.go#L49

File: pkg/expr/classic/classic_test.go [hunk 1]
```diff
@@ -169,6 +169,31 @@ func TestConditionsCmdExecute(t *testing.T) {
...
+			conditionsCmd: &ConditionsCmd{
+				Conditions: []condition{
+					{
+						QueryRefID: "A",
+						Reducer:    classicReducer("avg"),
+						Operator:   "and",
+						Evaluator:  &thresholdEvaluator{Type: "gt", Threshold: .5},
+					},
+				}},
+			resultNumber: func() mathexp.Number {
```
Comment (toddtreece) [inline pkg/expr/classic/classic_test.go:191]: what's the expected output from SSE if all series are null? it seems like it's currently this:

image

should it be an error?

Comment (kylebrandt) [reply pkg/expr/classic/classic_test.go:191]: Yes it should be Null, because the evaluator (IIRC) uses a nil number to be no data

**Comment by Dieterbe** [comment]
> I think probably so, it was an error on my part in translating the old alert behavior to into SSE.
> Previous code that this was meant to match: (...) 

[this line in particular] seems to confirm that no data used to mean "all series had no data", rather than "any series had no data". (https://github.com/grafana/grafana/blob/33b6f996e1a24e22a3f54f16a65a14635e407fb6/pkg/services/alerting/conditions/query.go#L105)

> @Dieterbe so you're saying we should make this a Graphite only thing? Are there any other data sources that behave similarly?

@santihernandezc that is not at all what I meant. I can only speak to Graphite, I'm not sure how this should be handled in other systems.

**Review by toddtreece — APPROVED** [review]
looks good to me 👍

**Pull request merged**